pilot/pkg/model/service.go

```diff
@@ -485,7 +485,8 @@ type IstioEndpoint struct {
 	Address string
 
 	// ServicePortName tracks the name of the port, this is used to select the IstioEndpoint by service port.
-	ServicePortName string
+	ServicePortName    string
+	ServicePortNameKey string
```

Comment:
I donot like this two meaning ambiguous

Reply:
yeah I don't like the implementation either. if we want to do this I'll refactor